### ISSUE TITLE
class Generator is only used by behavior clone?

### ISSUE THREAD

**Issue by onlytailei** [YunzhuLi/InfoGAIL#4]
class Generator in line 533 of [model.py](https://github.com/YunzhuLi/InfoGAIL/blob/master/wgail_info_1/models.py) is only used by behavior clone?

**Comment by onlytailei** [comment]
And how do get all of these pre-trained model params for fine-tuning?

**Comment by onlytailei** [comment]
And why you put the posterior predict as part of the TRPO reward? That is different from the original GAIL. In GAIL, the output_d is added to the reward with log. But here it is not.

**Comment by YunzhuLi** [comment]
1. class Generator in line 533 of model.py will be used by behavior_clone.py and drive.py
2. I have released one set of trained weights. Weights during the training process can be obtained using the released code & data.
3. Posterior prediction is the key to identify the latent structure, thus is needed as part of the reward. Here I used WGAN instead of regular GAN, thus output_d is added without log. Please refer to our paper for more details.

**Comment by onlytailei** [comment]
Thanks for your reply.
Then a quick question: there is a bias in the reward (2 every step). Is this what you call **reward augmentation** or just a bias? If this is the **reward augmentation**, should it be related to the collision condition?

**Comment by YunzhuLi** [comment]
Yes. It is the reward augmentation. Adding a reward (2 every step) encourages "staying alive" as long as possible, which can be considered as another way of discouraging collision, since collision will lead to the termination of that episode.

**Comment by onlytailei** [comment]
Thanks, nice work!
